Bottom: 5317d734747ab485224f309d80e317aa9791ebb5
Top:    8881f2b50660b44e97978286cd66d9b600810aa3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-24 02:07:21 +0000

Add some shared code for GL drawing


---

diff --git a/configure.ac b/configure.ac
index 18ab959..7873ad5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -261,6 +261,44 @@ $DBUS_PKG_ERRORS])]
 
 fi
 
+
+AC_MSG_CHECKING([for whether to use GL drawing])
+AC_ARG_ENABLE([gl],
+[  --enable-gl           Enable GL drawing (with GTK HID)],
+[],[enable_gl=no])
+
+AC_MSG_RESULT([$enable_gl])
+AM_CONDITIONAL(WITH_GL, test x$enable_gl = xyes)
+
+if test "x$enable_gl" = "xyes"; then
+	case " $with_gui " in
+		*\ gtk\ *) ;;
+		* ) AC_MSG_ERROR([GL drawing enabled but only works with the GTK HID.
+Either do not use --enable-gl or enable the gtk HID.])
+	esac
+	
+	# Check for pkg-config
+	AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+	if test "$PKG_CONFIG" = "no"; then
+		AC_MSG_ERROR([Cannot find pkg-config, make sure it is installed and in your PATH])
+	fi
+
+	PKG_CHECK_MODULES(DBUS, dbus-1 >= 0.61,
+		[saved_LIBS="$LIBS"
+		 LIBS="$LIBS $DBUS_LIBS"
+		 AC_CHECK_FUNCS(dbus_watch_get_unix_fd)
+		 LIBS="$saved_LIBS" ],
+		[AC_MSG_ERROR([Cannot find dbus-1 >= 0.61, install it and rerun ./configure
+Please review the following errors:
+$DBUS_PKG_ERRORS])]
+	)
+	DBUS_VERSION=`$PKG_CONFIG dbus-1 --modversion`
+	
+	AC_DEFINE([HAVE_DBUS], 1,
+		[Define to 1 if DBUS IPC is to be compiled in])
+
+fi
+
 AC_MSG_CHECKING([for which printer to use])
 AC_ARG_WITH([printer],
 [  --with-printer= 	  Specify the printer: lpr [[default=lpr]]],
@@ -505,7 +543,6 @@ $GLIB_PKG_ERRORS])]
 	)
 	GLIB_VERSION=`$PKG_CONFIG glib-2.0 --modversion`
 
-
 	# Check for GtkGLExt
 	PKG_CHECK_MODULES(GTKGLEXT, gtkglext-1.0 >= 1.0.0, , [AC_MSG_ERROR([
 *** Required version of gtkglext is not installed - please install first ***
diff --git a/src/Makefile.am b/src/Makefile.am
index 8109894..debc185 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -146,13 +146,17 @@ PCB_SRCS = \
 	hid/hidint.h 
 pcb_SOURCES = ${PCB_SRCS} core_lists.h
 
-EXTRA_pcb_SOURCES = ${DBUS_SRCS}
+EXTRA_pcb_SOURCES = ${DBUS_SRCS} ${GL_SRCS}
 DBUS_SRCS= \
 	dbus-pcbmain.c \
 	dbus-pcbmain.h \
 	dbus.h \
 	dbus.c
 
+GL_SRCS= \
+	hid/common/hidgl.c \
+	hid/common/hidgl.h
+
 BUILT_SOURCES = \
 	core_lists.h \
 	gpcb-menu.h \
@@ -267,6 +271,11 @@ BUILT_SOURCES+=	dbus-introspect.h
 
 endif
 
+# If we are building with GL support, we need some extra files
+if WITH_DBUS
+PCB_SRCS+=	${GL_SRCS}
+endif
+
 # If we are building on win32, then compile in some icons for the
 # desktop and application toolbar
 if WIN32
